### PR TITLE
Final map features

### PR DESCRIPTION
**Pull request by emaohn**
- Clears and rerender's route when appropriate (deleting, adding, editing)
- Set default behaviors for map orientation:
Nothing in map: Set center to default destination coordinates
Changes in travel objects in current display date: set bounds to that day's route
Changes in other locations (finalized, unfinalized): set bounds to all markers
- Don't include departure airport in bounds
- Other bug fixes with clicking and responsive behavior
- Clicking marker pops up mini info window with title of travel object
- Added comparator for lists of travel objects and individual travel objects

### CODE REVIEW

File: step-capstone/src/components/TravelObjects/TravelObject.js [hunk 1]
```diff
@@ -12,10 +12,8 @@ export default function TravelObject(props) {
     
     const handleClick = (event) => {
         setAnchorEl(event.currentTarget);
-        // if has a placeId, zoom to marker on map
```
Comment (ganewman42) [inline step-capstone/src/components/TravelObjects/TravelObject.js:15]: Curious: What made this check no longer necessary?

File: step-capstone/src/components/Utilities/Map.js [hunk 2]
```diff
@@ -40,24 +37,41 @@ class MapComponent extends React.Component {
...
+
+      this.zoomToMarker(marker);
+    } else if (!sameTravelObjectList(prevProps.displayDate.events, this.props.displayDate.events) && this.props.displayDate.events.length !== 0) {
+      // if changes to current day's events and isn't empty, zoom to todays events
       this.clearMap()
       this.drawMap();
       this.googleMap.fitBounds(this.getTodaysBounds());
-    } else if (prevProps.finalized !== this.props.finalized || prevProps.unfinalized !== this.props.unfinalized) {
+    } else if (!sameTravelObjectList(prevProps.finalized, this.props.finalized) || !sameTravelObjectList(prevProps.unfinalized, this.props.unfinalized)) {
+      // any other changes to finalized or unfinalized zoom out to encompass all markers.
```
Comment (ganewman42) [inline step-capstone/src/components/Utilities/Map.js:48]: Nice work on the comments.

File: step-capstone/src/scripts/HelperFunctions.js [hunk 3]
```diff
@@ -21,20 +21,61 @@ export const isValid = function (date) {
...
+    }
+  }
+  return true;
+}
+
+export const sortTravelObjectsByDate = function (list) {
+  return list.sort(travelObjectStartDateComparator)
+}
+
+const equalsTravelObject = function (obj1, obj2) {
```
Comment (ganewman42) [inline step-capstone/src/scripts/HelperFunctions.js:61]: Nice work, this a good concept of equality.

File: step-capstone/src/components/TravelObjects/TravelObject.js (hunk 1 above)
Comment (emaohn) [reply step-capstone/src/components/TravelObjects/TravelObject.js:15]: I have a check in the map component that only zooms to the marker if it exists in the map (key is placeId)

Comment (emaohn) [reply step-capstone/src/components/TravelObjects/TravelObject.js:15]: Actually never mind, I did need it--thanks for catching!

**Pull request merged**